web/views.py

```diff
@@ -299,6 +300,35 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
         return context
 
 
+class ParticipantEmailUnsubscribeView(generic.View):
+    template_name = "web/unsubscribe.html"
+
+    def get(self, request, username, *args, **kwargs):
+        return render(request, self.template_name, {"username": username})
+
+    def post(self, request, username, token, *args, **kwargs):
+        user = User.objects.get(username=username)
+
+        if user.check_token(token):
+            User.objects.filter(pk=user.id).update(
+                email_new_studies=False,
+                email_next_session=False,
+                email_study_updates=False,
+                email_response_questions=False,
+            )
+            messages.info(request, f"{username} has been unsubscribed.")
+        else:
+            email_pref_url = reverse("web:email-preferences")
+            messages.error(
+                request,
+                mark_safe(
+                    f'{username} could not be unsubscribed. Click <a href="{email_pref_url}">here</a> to update your email preferences.'
+                ),
+            )
+
+        return redirect("web:home")
```

Comment:
Could we maybe present the unsubscribe success/failure message on the user's email preferences page, rather than the home page? Or does the token-based authentication not allow us to do that?
If it is possible, it might be nice to remind the user what the email categories are after they've clicked the unsubscribe-all link, in case they want to change it.

Reply:
@okaycj NM, I see in your updated PR description that you chose to redirect to home because it doesn't require being logged in. Makes sense!